src/stk/calculators/base_calculators.py

```diff
@@ -52,6 +52,37 @@ class MoleculeCalculator(Calculator):
 
     """
 
+    def _cache_result(self, fn, mol):
```

Comment:
Could you explain why this is in the abstract base calculator class and not in _MolecularCalculator as an implementation detail?
It looks to me like this function relies on self.is_caching() and self.is_in_cache() being present, however, they are not implemented in the base call.

Reply:
Methods in an abstract base class are not expected to be called. This ABC provides provides a customizable algorithm, with the custmization provided by the is_caching(), is_in_cache() calls. A subclass will need to implement the virtual methods and will be allowed to call this algorithm. So specifically, this is ABC does not expect a subclass to implement an algoirithm for _cache_restul. it exepcts sublcassses to use this algorith, but it allows it customization via the virtual method calls.